File: dehaze-python/app/models/entity/sys_rating.py
from datetime import datetime

from sqlalchemy import JSON, BigInteger, DateTime, String
from sqlalchemy.dialects import mysql as mysql_types
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel, SoftDeleteMixin


class SysRating(BaseModel, SoftDeleteMixin):
    __tablename__ = "sys_rating"
    __table_args__ = {"comment": "处理结果评分评价表"}

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    user_id: Mapped[int] = mapped_column(BigInteger, nullable=False, comment="评价用户ID")
    pred_log_id: Mapped[int] = mapped_column(BigInteger, nullable=False, comment="关联处理日志ID")
    algorithm_id: Mapped[int] = mapped_column(BigInteger, nullable=False, comment="算法ID")
    rating: Mapped[int] = mapped_column(mysql_types.TINYINT, nullable=False, comment="评分(1-5星)")
    comment: Mapped[str | None] = mapped_column(String(500), nullable=True, comment="评价文字")
    tags: Mapped[list | None] = mapped_column(JSON, nullable=True, comment="评价标签（JSON数组）")
    image_urls: Mapped[list | None] = mapped_column(
        JSON, nullable=True, comment="截图URL（JSON数组）"
    )
    is_anonymous: Mapped[int] = mapped_column(
        mysql_types.TINYINT, nullable=False, default=0, comment="是否匿名(0:否;1:是)"
    )
    is_hidden: Mapped[int] = mapped_column(
        mysql_types.TINYINT, nullable=False, default=0, comment="是否隐藏(0:否;1:是)"
    )
    admin_reply: Mapped[str | None] = mapped_column(
        String(2000), nullable=True, comment="管理员回复内容"
    )
    reply_time: Mapped[datetime | None] = mapped_column(
        DateTime, nullable=True, comment="管理员回复时间"
    )

File: dehaze-go/internal/model/sys_rating.go
package model

import "time"

type SysRating struct {
	BaseModel
	UserID       int64      `gorm:"column:user_id;type:bigint;not null;index:idx_user_id;comment:评价用户ID" json:"userId"`
	PredLogID    int64      `gorm:"column:pred_log_id;type:bigint;not null;uniqueIndex:uk_pred_log_id;comment:关联处理日志ID" json:"predLogId"`
	AlgorithmID  int64      `gorm:"column:algorithm_id;type:bigint;not null;index:idx_algorithm_id;comment:算法ID" json:"algorithmId"`
	Rating       int8       `gorm:"column:rating;type:tinyint;not null;index:idx_rating;comment:评分(1-5星)" json:"rating"`
	Comment      string     `gorm:"column:comment;type:varchar(500);comment:评价文字" json:"comment"`
	Tags         string     `gorm:"column:tags;type:json;comment:评价标签（JSON数组）" json:"tags"`
	ImageUrls    string     `gorm:"column:image_urls;type:json;comment:截图URL（JSON数组）" json:"imageUrls"`
	IsAnonymous  int8       `gorm:"column:is_anonymous;type:tinyint;not null;default:0;comment:是否匿名(0:否;1:是)" json:"isAnonymous"`
	IsHidden     int8       `gorm:"column:is_hidden;type:tinyint;not null;default:0;comment:是否隐藏(0:否;1:是)" json:"isHidden"`
	AdminReply   string     `gorm:"column:admin_reply;type:varchar(2000);comment:管理员回复内容" json:"adminReply"`
	ReplyTime    *time.Time `gorm:"column:reply_time;type:datetime;comment:管理员回复时间" json:"replyTime"`
	Deleted      int8       `gorm:"column:deleted;type:tinyint;not null;default:0;comment:逻辑删除标识" json:"deleted"`
}

func (SysRating) TableName() string {
	return "sys_rating"
}
